web/source/kmwuibutton.ts

```diff
@@ -199,100 +200,104 @@ if(!window['keyman']['ui']['name']) {
+     *
      * @param       {?string=}  _name     current keyboard name
-     */    
+     */
     ui._ShowKeyboardButton = function(_name)
     {
       var kbdName = keymanweb['getActiveKeyboard'](), kbdId=document.getElementById("KMW_Keyboard");
       if(arguments.length > 0) kbdName = _name;
       if(kbdId)
       {
         if((kbdName == '') || keymanweb['isCJK']())
-        { 
+        {
           kbdId.className='kmw_disabled';
-        } 
+        }
         else
         {
           let osk = keymanweb.osk;
           kbdId.className = osk && osk['isEnabled']() ? 'kmw_show' : 'kmw_hide';
         }
       }
-    }  
-  
+    }
+
     ui.registerEvents = function() {
       let osk = keymanweb.osk;
       if(!osk) {
         return;
       }
       /**
        * UI Functions called by KeymanWeb or OSK
-       */     
-      osk['addEventListener']('show', function(oskPosition) { 
+       */
+      osk['addEventListener']('show', function(oskPosition) {
         var t=keymanweb['getLastActiveElement']();
         if(t)
-        {  
+        {
           if(!oskPosition['userLocated'])
           {
             oskPosition['x'] = util['getAbsoluteX'](t);
             oskPosition['y'] = util['getAbsoluteY'](t)+t.offsetHeight;
           }
-        } 
-        
+        }
+
         ui._ShowKeyboardButton();
-        return oskPosition; 
-      }); 
+        return oskPosition;
+      });
 
-      /* TODO: why is this still needed??? Does it actually do anything?? */ 
-      osk['addEventListener']('hide', function(hiddenByUser) { 
+      /* TODO: why is this still needed??? Does it actually do anything?? */
+      osk['addEventListener']('hide', function(hiddenByUser) {
         if((arguments.length > 0) && hiddenByUser)
         {
           var _a = document.getElementById('KMW_Keyboard');
           if(_a) _a.className = 'kmw_hide';
-        }    
-      }); 
+        }
+      });
     }
-    
+
     /**
      * Show or hide the OSK (always visible for CJK keyboards)
-     * 
+     *
      * @param       {Object}    _anchor     anchor element (?)
-     * @return      {boolean}   
-     **/   
-    ui._ShowKeymanWebKeyboard = function(_anchor) 
-    { 
+     * @return      {boolean}
+     **/
+    ui._ShowKeymanWebKeyboard = function(_anchor)
+    {
       var kbdId=document.getElementById("KMW_Keyboard");
       let osk = keymanweb.osk;
 
-      if((kbdId.className!='kmw_disabled') && osk && osk['show']) 
+      if((kbdId.className!='kmw_disabled') && osk && osk['show'])
       {
         if(osk['isEnabled']()) osk['hide'](); else osk['show'](true);
       }
       if(window.event) window.event.returnValue=false;
 
-      keymanweb['focusLastActiveElement']();    
-      return false;    
+      keymanweb['focusLastActiveElement']();
+      return false;
     }
 
     /**
      * Initialize Button User Interface
-     **/   
+     **/
     ui['initialize'] = ui.Initialize = function() {
+      window.clearTimeout(ui.initTimer);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout

> If the parameter provided does not identify a previously established action, this method does nothing. 

We technically don't _need_ the suggested change here; extra timeout calls are just no-ops.   That said... clear bookkeeping probably is better.

Reply:
Yeah, TIL that timer IDs are never reused (per https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#return_value). So you are right. However, clearing it to `null` helps in case we want to ever know if we set a timeout in the future. So being tidy is probably good policy anyway.